pwnlib/rop/rop.py

```diff
@@ -1206,6 +1206,92 @@ def search(self, move = 0, regs = None, order = 'size'):
 
         return result
 
+    def ret2csu(self, edi, rsi, rdx, rbx=0, rbp=0, r12=0, r13=0, r14=0, r15=0, call=None):
+        """Build a ret2csu ROPchain
+
+        Arguments:
+            edi, rsi, rdx: Three primary registers to populate
+            rbx, rbp, r12, r13, r14, r15: Optional registers to populate
+            call: Pointer to the address of a function to call during
+                second gadget. If None then use the address of _fini in the
+                .dynamic section. .got.plt entries are a good target. Required
+                for PIE binaries.
+
+        Examples:
+            >>> rop = ROP(elf64_noPIE)
+            >>> rop.ret2csu(edi, rsi, rdx)
+            >>> print rop.dump()
+
+            >>> rop = ROP(elf64_PIE)
+            >>> rop.ret2csu(edi, rsi, rdx, call=elf.got.read)
+            >>> print rop.dump()
```

Comment:
Same here, this needs to be an actual test.  I'd recommend picking a binary from `pwnlib.data.elf`.